libopendroneid/opendroneid.c

```diff
@@ -378,10 +378,18 @@ int encodeAuthMessage(ODID_Auth_encoded *outEncoded, ODID_Auth_data *inData)
         return ODID_FAIL;
 
     if (inData->DataPage == 0) {
-        if (inData->LastPageIndex >= ODID_AUTH_MAX_PAGES ||
-            inData->Length > MAX_AUTH_LENGTH)
+        if (inData->LastPageIndex >= ODID_AUTH_MAX_PAGES)
             return ODID_FAIL;
 
+#if (ODID_AUTH_MAX_PAGES <= 11) 
+        if (inData->Length > MAX_AUTH_LENGTH)
+            return ODID_FAIL;
+#else
+        if (inData->LastPageIndex > 10 &&
+            inData->Length != UINT8_MAX)
+            return ODID_FAIL;
+#endif
+
```

Comment:
I would propose to fix this instead with the following code snippet:
```
#if (MAX_AUTH_LENGTH < UINT8_MAX)
        if (inData->Length > MAX_AUTH_LENGTH)
            return ODID_FAIL;
#endif
```

Reply:
Better. Done.